patches/minecraft/net/minecraft/util/IChatComponent.java.patch

```diff
@@ -8,3 +8,44 @@
      String func_150254_d();
  
      List<IChatComponent> func_150253_a();
+@@ -57,7 +56,7 @@
+ 
+                         for (JsonElement jsonelement : jsonarray1)
+                         {
+-                            IChatComponent ichatcomponent2 = this.deserialize(jsonelement, jsonelement.getClass(), p_deserialize_3_);
++                            IChatComponent ichatcomponent2 = p_deserialize_3_.deserialize(jsonelement, IChatComponent.class);
```

Comment:
Gson-doc for <code>deserialize()</code> suggests to do it this way.
It's also easier to hook into the deserialization process. Let's say you have a bunch of JsonObjects describing IChatComponents with common extra-tags (for whatever reason...) and you want to cache them. If the deserialization process is rerouted over the context, you can easily implement the cache in a custom context and use the Gson-object as fallback. If however <code>this.deserialize()</code> is called you have to extend the deserializer and construct a new Gson-object for every cache which is certainly much more inconvenient.. ;-).
Anyway, the type currently passed to <code>deserialize()</code> (<code>jsonelement.getClass()</code>) is horribly wrong even if it is ignored by the deserializer ;-))